File: app/src/main/java/com/chad/baserecyclerviewadapterhelper/entity/MySection.java
package com.chad.baserecyclerviewadapterhelper.entity;

import com.chad.library.adapter.base.entity.JSectionEntity;

/**
 * https://github.com/CymChad/BaseRecyclerViewAdapterHelper
 * <p>
 * java请自定义类，继承于JSectionEntity抽象类。封装一遍自己的数据类
 * <p>
 * kotlin，数据类请直接实现SectionEntity接口即可，无需封装。
 */
public class MySection extends JSectionEntity {
    private boolean isHeader;
    private Object  object;

    public MySection(boolean isHeader, Object object) {
        this.isHeader = isHeader;
        this.object = object;
    }

    public Object getObject() {
        return object;
    }

    @Override
    public boolean isHeader() {
        return isHeader;
    }

}

File: kotlinapp/src/main/java/com/allen/kotlinapp/entity/MySection.kt
package com.allen.kotlinapp.entity

import com.chad.library.adapter.base.entity.SectionEntity

/**
 * 文 件 名: MySection
 * 创 建 人: Allen
 * 创建日期: 2017/6/13 14:18
 * 修改时间：
 * 修改备注：
 */
class MySection : SectionEntity<Video> {
    var isMore: Boolean = false

    constructor(isHeader: Boolean, header: String, isMroe: Boolean) : super(isHeader, header) {
        this.isMore = isMroe
    }

    constructor(t: Video) : super(t) {}
}
